development/fitness-functions/rules/prevent-deprecated-imports.ts

```diff
@@ -0,0 +1,43 @@
+import { filterDiffLineAdditions } from '../common/shared';
+
+// Note: Remove this file once we have migrated all deprecated components to the design system
+
+const DEPRECATED_COMPONENT_PATHS = [
+  'component-library/avatar-base',
+  'component-library/avatar-network',
+  'component-library/avatar-token',
+  'component-library/badge-wrapper',
+  'component-library/banner-alert',
+  'component-library/box',
+  'component-library/button',
+  'component-library/button-base',
+  'component-library/button-link',
+  'component-library/button-primary',
+  'component-library/button-secondary',
+  'component-library/checkbox',
+  'component-library/form-text-field',
+  'component-library/icon',
+  'component-library/modal-header',
+  'component-library/popover',
+  'component-library/text',
+  'component-library/text-field-search',
+  'component-library/textarea',
```

Comment:
non-blocking: should we also add deprecated components from the ui folder? e.g.
```
'ui/box',
'ui/button',
'ui/check-box',
'ui/typography',
'ui/form-field',
'ui/chip',
'ui/text-field',
'ui/identicon',
'ui/popover',
'ui/site-origin',
'ui/url-icon',
'ui/textarea',
'ui/menu',
'ui/icon-with-fallback',
'ui/icon-border',
'ui/callout',
'ui/actionable-message'
```

Got this list from searching `@deprecated` in `ui/components/ui`

<img width="363" height="283" alt="Image" src="https://github.com/user-attachments/assets/efcfe0a2-8ecb-48e7-9775-d55334b34348" />

Reply:
Updated!